cpp/src/arrow/adapters/orc/adapter.cc

```diff
@@ -217,6 +217,8 @@ class ORCFileReader::Impl {
 
   int64_t NumberOfRows() { return static_cast<int64_t>(reader_->getNumberOfRows()); }
 
+  int64_t NumberOfRows(int64_t stripe) { return stripes_[stripe].num_rows; }
```

Comment:
Should we rename it to `NumberOfRowsInStripe` to avoid unintentional overloading?

Reply:
In addition, it is better to validate `stripe` and throw if out of range.

Reply:
Well maybe it will be clearer to rename it NumberOfRowsInStripe indeed.
What kind of exception would you find appropriate, `liborc::InvalidArgument` ?

Reply:
It seems that the orc adapter already throws exceptions from liborc. Maybe it is good enough to throw `liborc::InvalidArgument` for now.

Reply:
@kou Can you advise if we need to be more consistent to the project style which returns status instead of throwing exception?

Reply:
We shouldn't throw exceptions in Arrow. Also, I don't think there's a need for checking the input value: the number of stripes is readily available.